cmd/yurt-manager/app/client/client.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"net/http"
+	"sync"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/openyurtio/openyurt/cmd/yurt-manager/app/config"
+)
+
+const serviceAccountPrefix = "yurt-manager-"
+
+var clientStore = &struct {
+	clientByName map[string]client.Client
+	lock         sync.Mutex
+}{
+	clientByName: make(map[string]client.Client),
+	lock:         sync.Mutex{},
+}
+
+var configStore = &struct {
+	configByName map[string]*rest.Config
+	lock         sync.Mutex
+}{
+	configByName: make(map[string]*rest.Config),
+	lock:         sync.Mutex{},
+}
+
+func GetConfigByControllerNameOrDie(mgr manager.Manager, controllerName string) *rest.Config {
+
+	namespace := config.WorkingNamespace
+
+	// if controllerName is empty, return the base client of manager
+	if controllerName == "" {
+		return mgr.GetConfig()
+	}
+
+	configStore.lock.Lock()
+	defer configStore.lock.Unlock()
+
+	if cfg, ok := configStore.configByName[controllerName]; ok {
+		return cfg
+	}
+
+	// get base config
+	baseCfg := mgr.GetConfig()
+	baseClient, err := kubernetes.NewForConfig(baseCfg)
```

Comment:
baseClient can be shared by all clients, so only one baseClient should be created for all tokens.

Reply:
done